applications/vanilla/views/categories/discussions.php

```diff
@@ -36,7 +36,38 @@
 
                 </div>
 
-            <?php endif; ?>
+            <?php else:
+
+                if ($Category->CountCategories > 0) { ?>
+
+                    <div class="CategoryBox Category-<?php echo $Category->UrlCode; ?>">
+                        <?php echo getOptions($Category); ?>
+                        <h2 class="H">
+                            <?php
+                                echo anchor(htmlspecialchars($Category->Name), categoryUrl($Category));
+                                Gdn::controller()->EventArguments['Category'] = $Category;
+                                Gdn::controller()->fireEvent('AfterCategoryTitle');
+                            ?>
+                        </h2>
+                    </div>
```

Comment:
This looks like the exact same code. Can you refactor so it's just one block? It seems alike you can take the main if statement out altogether and then just wrap an if statement around the table output no?